common/type_system/TypeSystem.cpp

```diff
@@ -1590,7 +1590,7 @@ bool TypeSystem::typecheck_and_throw(const TypeSpec& expected,
       if (!got) {
         success = false;
       } else {
-        if (*got != tag.value) {
+        if (!tc(tag.value, *got)) {
```

Comment:
I really don't remember why it was like this in the first place... I can't think of any reason why. At some point maybe I thought that tag values wouldn't always be types? Either way I think this is a good change.

It also looks like tag stuff is just missing from the `TypeSystem::lowest_common_ancestor` function (it will just strip away tags right now?)